appinventor/appengine/src/com/google/appinventor/client/widgets/dnd/DragSourceSupport.java

```diff
@@ -119,6 +122,9 @@ public DragWidgetPopup(Widget w) {
   private int dragX;
   private int dragY;
 
+  // List of MockTableArrangements to which mouse click position is passed
+  private static ArrayList<MockTableArrangement> tableArrangementsList;
```

Comment:
I have some concerns about storing the table arrangements as a static variable. This means that if I have multiple projects that I have opened, all of the arrangements across all of those projects will be present here, so the check is linear in the number of arrangements whereas the solution could probably be redesigned to be constant time.

Changing the drag support source in this way also violates separation of concerns, i.e., it shouldn't need to know about MockTableArrangements at all. For example, if you want to track whether the user initiated a mouse down event in a cell you should add a handler to the appropriate Widget/Element.

Reply:
@ewpatton 
If a new Mouse Down Event handler is implemented for MockTableArrangement component then we will not have any functionality problem but there will the following problem:

When user clicks inside an empty cell of table layout, there will be a border around the cell. Now if the user clicks outside the table layout then the border will not be hidden, since in this case mouse down event will not be triggered (the table layout will not know things happening outside it). So if user clicks inside and doesn't paste any item inside table layout then the border will be shown all the time until the user again interacts with the table layout.

So what should be the behaviour now,
1.Should I not show a border around the cell when the user clicks inside the cell (if this is the case then we don't even require any mouse event handlers to fix the bug) or 
2.Leave as it is i.e. show the border all the time until further action with the table layout.
3.Show the border around the cell for a constant time of 3-4 seconds.
or do you have any other alternative

Reply:
I am assuming that it is ok to create two static integers in drag source support to store location of click, correct me if it is not allowed. If it is not allowed then there would be one more problem when user pastes an item onto a already existing child inside table layout. In this case also mouse event will not be triggered due to which we can't find the target cell.